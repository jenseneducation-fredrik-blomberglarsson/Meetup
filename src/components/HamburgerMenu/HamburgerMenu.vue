<template>
  <nav class="hamburgerMenu" v-if="getBurgerToggle">
    <img class="closeBtn" src="../../assets/close.svg" alt="close" @click="changeThisBurgerToggle" />
    <router-link v-on:click.native="changeThisBurgerToggle" to="/" class="router-link">Home</router-link>
    <hr class="hr" />
    <router-link
      v-on:click.native="changeThisBurgerToggle"
      to="/meetups"
      class="router-link"
    >Upcoming Meetups</router-link>
    <hr class="hr" />
    <router-link
      v-on:click.native="changeThisBurgerToggle"
      to="/mymeetups"
      class="router-link"
    >My Meetups</router-link>
  </nav>
</template>

<script>
import { mapActions, mapGetters } from "vuex";
export default {
  name: "HamburgerMenu",
  methods: {
    ...mapActions(["changeThisBurgerToggle"]),
  },
  computed: {
    ...mapGetters(["getBurgerToggle"]),
  },
};
//-webkit-linear-gradient(#03a9f4, #f441a5, #ffeb3b, #03a9f4);
</script>

<style>
@import url("https://fonts.googleapis.com/css2?family=Permanent+Marker&display=swap");
.hamburgerMenu {
  font-family: "Permanent Marker", cursive;
  height: 100vh;
  width: 100%;
  position: fixed;
  top: 0;
  left: 0;
  padding: 0;
  background: linear-gradient(#03a9f4, #f441a5, #ffeb3b, #03a9f4);
  background-size: 400% 400%;
  animation: change 10s ease-in-out infinite;
  display: flex;
  flex-direction: column;
  justify-content: center;
  z-index: 9999;
  cursor: default;
}
@keyframes change {
  0% {
    background-position: 0 100%;
  }
  50% {
    background-position: 100% 50%;
  }
  100% {
    background-position: 0 100%;
  }
}
.router-link {
  padding: 1rem;
  display: flex;
  flex-direction: column;
  align-items: center;
  font-size: 3rem;
  text-decoration: none;
  color: white;
}
.hr {
  width: 20%;
}
.closeBtn {
  position: absolute;
  top: 1em;
  right: 2em;
  cursor: pointer;
  background: white;
  width: 1.5em;
  padding: 1rem;
  border-radius: 50px;
}
</style>