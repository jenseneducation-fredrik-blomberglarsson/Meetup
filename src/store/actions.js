import { getMeetups } from "../data/meetupData";

const actions = {
  async fetchMeetups(context) {
    const data = await getMeetups();
    context.commit("setMeetups", data);
  }
}

export default actions;