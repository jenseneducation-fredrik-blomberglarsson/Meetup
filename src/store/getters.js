const getters = {
  getMeetups: state => {
    return state.meetups;
  },
  getJoinedMeetups: state => {
    return state.joinedMeetups;
  },
  getPreviousMeetups: state => {
    return state.previousMeetups;
  },
  getMeetupToggle: state => {
    return state.meetupToggle;
  },
  getBurgerToggle: state => {
    return state.burgerToggle;
  }
}


export default getters;