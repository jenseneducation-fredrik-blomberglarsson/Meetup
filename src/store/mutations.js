const mutations = {
  setMeetups(state, meetups) {
    state.meetups = meetups;
  }
}

export default mutations;