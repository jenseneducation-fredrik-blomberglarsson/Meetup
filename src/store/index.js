import Vue from 'vue'
import Vuex from 'vuex'
import actions from './actions.js';
import mutations from './mutations.js';
import getters from './getters.js';

Vue.use(Vuex)

//Global state
export default new Vuex.Store({
  state: {
    meetups: [],
    joinedMeetups: [],
    previousMeetups: [],
    meetupToggle: false,
  },

  actions: actions,
  mutations: mutations,
  getters: getters
});