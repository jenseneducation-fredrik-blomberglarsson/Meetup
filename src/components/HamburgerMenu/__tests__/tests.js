import { shallowMount, mount, createLocalVue, RouterLinkStub } from '@vue/test-utils';
import Vuex from 'vuex';
import store from '../../../store/index.js';
import HamburgerMenu from '@/components/HamburgerMenu/HamburgerMenu.vue';
import VueRouter from 'vue-router';

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(VueRouter);

describe('Menu is opened', () => {
  test('Test so that everything renders properly', async () => {
    const state = {
      burgerToggle: true
    };
    const getters = {
      getBurgerToggle: state => {
        return state.burgerToggle;
      }
    };
    const store = new Vuex.Store({
      getters,
      state
    });

    const wrapper = shallowMount(HamburgerMenu, { store, localVue });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('Test that HamburgerMenu component is rendered when burgerToggle is true', () => {
    const state = {
      burgerToggle: true
    };
    const getters = {
      getBurgerToggle: state => {
        return state.burgerToggle;
      }
    };
    const store = new Vuex.Store({
      getters,
      state
    });
    const wrapper = shallowMount(HamburgerMenu, {
      store,
      localVue
    });

    expect(wrapper.find('.hamburgerMenu').exists()).toBe(true);
  });

  test('Test that HamburgerMenu component is not rendered when burgerToggle is false', () => {
    const wrapper = shallowMount(HamburgerMenu, {
      store,
      localVue
    });

    expect(wrapper.find('.hamburgerMenu').exists()).toBe(false);
  });

  test('Check that close button exists', async () => {
    const state = {
      burgerToggle: true
    };
    const getters = {
      getBurgerToggle: state => {
        return state.burgerToggle;
      }
    };
    const store = new Vuex.Store({
      getters,
      state
    });
    const wrapper = shallowMount(HamburgerMenu, {
      store,
      localVue
    });

    const closeBtn = wrapper.find('.closeBtn');
    expect(closeBtn.exists()).toBe(true);
  });

  test('Test that close button closes the menu', async () => {
    const state = {
      burgerToggle: true
    };
    const actions = {
      changeThisBurgerToggle: jest.fn()
    };
    const getters = {
      getBurgerToggle: state => {
        return state.burgerToggle;
      }
    };
    const store = new Vuex.Store({
      actions,
      getters,
      state
    });
    const wrapper = shallowMount(HamburgerMenu, {
      store,
      localVue
    });

    const closeBtn = wrapper.find('.closeBtn');
    await closeBtn.trigger('click');

    expect(actions.changeThisBurgerToggle).toHaveBeenCalled();
  });

  test('Test that router-links also closes menu', async () => {
    const state = {
      burgerToggle: true
    };
    const actions = {
      changeThisBurgerToggle: jest.fn()
    };
    const getters = {
      getBurgerToggle: state => {
        return state.burgerToggle;
      }
    };
    const store = new Vuex.Store({
      actions,
      getters,
      state
    });
    const wrapper = shallowMount(HamburgerMenu, {
      store,
      localVue
    });

    const routerLink = wrapper.find('.router-link');
    await routerLink.trigger('click');

    expect(actions.changeThisBurgerToggle).toHaveBeenCalled();
  })

  test('test if router-link for front-page exists in component HamburgerMenu', () => {
    const state = {
      toggleBurger: true
    };
    const getters = {
      getBurgerToggle: state => {
        return state.toggleBurger;
      }
    };
    const store = new Vuex.Store({
      getters,
      state
    });

    const wrapper = mount(HamburgerMenu, {
      store,
      localVue,
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    expect(wrapper.findComponent(RouterLinkStub).props().to).toBe('/');
  });
})