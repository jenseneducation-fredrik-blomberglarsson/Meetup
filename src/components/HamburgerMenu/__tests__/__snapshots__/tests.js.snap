// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu is opened Test so that everything renders properly 1`] = `
<nav
  class="hamburgerMenu"
>
  <img
    alt="close"
    class="closeBtn"
    src="../../assets/close.svg"
  />
   
  <router-link-stub
    ariacurrentvalue="page"
    class="router-link"
    event="click"
    tag="a"
    to="/"
  >
    Home
  </router-link-stub>
   
  <hr
    class="hr"
  />
   
  <router-link-stub
    ariacurrentvalue="page"
    class="router-link"
    event="click"
    tag="a"
    to="/meetups"
  >
    Upcoming Meetups
  </router-link-stub>
   
  <hr
    class="hr"
  />
   
  <router-link-stub
    ariacurrentvalue="page"
    class="router-link"
    event="click"
    tag="a"
    to="/mymeetups"
  >
    My Meetups
  </router-link-stub>
</nav>
`;
